silkrpc/protos/eth.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+// option java_multiple_files = true;
+// option java_package = "io.grpc.examples.rpcdaemon";
+// option java_outer_classname = "EthProto";
+// option objc_class_prefix = "ETH";
+
+// The primary namespace/package in the generated code
+package rpcdaemon;
+
+// Serves the eth_ RPC routines among other things
+service EthService {
+    // Returns block data as per request
+    rpc GetBlock(GetBlockRequest) returns(GetBlockResponse) {
+//        option (google.api.http) = {
+//            post: "/v1/example/echo"
+//            body: "*"
+//        };
+    }
+    // Returns transaction data as per request
+    rpc GetTransaction(GetTransactionRequest) returns(GetTransactionResponse) {
+    }
+}
+
+// Structure used to handle requests for eth_getBlock*, eth_getUncle, etc.
+message GetBlockRequest {
+  uint64 first = 1;
+  uint64 last = 3;
+  uint64 step = 5;
+  string hash = 7;
+  bool hashes_only = 9;
+  bool include_receipts = 11;
+}
+
+// The returned data from GetBlock related requests
+message GetBlockResponse { string message = 1; }
+
+// Structure used to handle requests for eth_getTrans* related calls
+message GetTransactionRequest {
+  uint64 block = 1;
+  uint64 tx_id = 3;
+  string hash = 5;
+}
+
+// The returned data from eth_getTrans* related requests
+message GetTransactionResponse { string message = 1; }
```

Comment:
The goals of [Silkrpc](https://github.com/torquem-ch/silkrpc/blob/master/docs/ARCHITECTURE.md) project are _both_ 1) producing a C++ implementation of the existing `ETH JSON RPC` interface IF#2 separated from but connected to Turbo-Geth/Silkworm Core via IF#1 _and_ 2) exploring the possibility to define an improved extended interface IF#3, probably binary for increased performance.

This `eth.proto` file and the related build files plus code (test_client, main) belong to the goal 2), which is still under discussion in terms of design, i.e. what IF#3 interface should offer exactly and how. 

So IMHO the content of this PR could stay in a dedicated branch until we finish the exploration of 2) and the design is finished with related decisions taken (e.g. which primitives are exposed through IF#3? do we need two different daemons?), just in order to maintain the `master` branch as clean as possibile.